core/trino-main/src/main/java/io/trino/sql/planner/QueryPlanner.java

```diff
@@ -1156,7 +1156,7 @@ private GroupingSetsPlan planGroupingSets(PlanBuilder subPlan, QuerySpecificatio
             groupingSetMappings.put(output, input);
         }
 
-        Map<ScopeAware<Expression>, Symbol> complexExpressions = new HashMap<>();
+        Map<ScopeAware<Expression>, Symbol> complexExpressions = new LinkedHashMap<>();
```

Comment:
Does this affect the order of expressions in the plan? If I recall correctly, this is just for doing some mapping later, but they are not used for driving the order in which things get added to the plan.

Reply:
There is
```
        // combine (cartesian product) with complex expressions
        List<List<Symbol>> groupingSets = sets.stream()
                .map(set -> ImmutableList.<Symbol>builder()
                        .addAll(set)
                        .addAll(complexExpressions.values())
                        .build())
                .collect(toImmutableList());
```
below.

Reply:
yes, the code that @sopel39 pasted uses `complexExpressions.values() `which is not deterministict for HashSet